src/mlpack/methods/decision_tree/best_binary_numeric_split_impl.hpp

```diff
@@ -25,9 +25,14 @@ double BestBinaryNumericSplit<FitnessFunction>::SplitIfBetter(
     const WeightVecType& weights,
     const size_t minimumLeafSize,
     const double minimumGainSplit,
+    const size_t maximumDepth,
     arma::Col<typename VecType::elem_type>& classProbabilities,
     AuxiliarySplitInfo<typename VecType::elem_type>& /* aux */)
 {
+  // First sanity check: if we have reached maximum depth, we can't split.
+  if (maximumDepth == 1)
+    return DBL_MAX;
```

Comment:
Honestly I think it would be better to handle this condition at a higher level (in `DecisionTree::Train()` instead).  If we handle it here, then every split policy (including `AllCategoricalSplit` which I don't think you changed) has to have this check in it.  If `maximumDepth == 1` in the regular tree, you can just terminate training and call it a leaf.  We can't do this with `minimumLeafSize`, though, because different split strategies will have different conditions for that (since they may have different numbers of children).

Reply:
Now I have just added this for sanity check.

Reply:
Ah, but can we remove it?  I like sanity checks, but the problem with this sanity check is that it requires changing the API of any numeric split policy.  So if anyone wants to implement a new numeric split policy, they now have to handle this `maximumDepth` parameter, which they only need to sanity check.  In my view that's unnecessary overhead and I'd prefer to do the checking at the level of the decision tree only. :+1: